### DISCUSSION TITLE
umi4，qiankun没有导出useModel

### DISCUSSION THREAD

**Discussion by luanwulin** [umijs/umi#8416] [Q&A] [answered]
umi4.0.5版本，没有导出useModel
![image](https://user-images.githubusercontent.com/16553854/177494858-4009325d-4eec-44f2-b469-190643719b94.png)

**Comment by fz6m** [comment, marked as answer]
使用 `useModel` 要使用 `@umijs/max` 或者加载 model 插件：

```bash
  pnpm add @umijs/plugins
```

```ts
// .umirc.ts

export default {
  plugins: [
    '@umijs/plugins/dist/model'
  ],
  model: {}
};

```

**Reply by luanwulin** [reply]
@fz6m 找遍了源码，，总算解决了。用max命令的话不行，就会出现上面的报错，只能用umi命令来启动，然后在plugins里面去注册model插件

**Reply by nopapername** [reply]
这个升级版本升级又是一套写法，就不能平滑升级么。。。

**Comment by lbyupup** [comment]
按照 @fz6m 提示的配置后报了下面这个错误，有大佬帮忙看看吗
![image](https://user-images.githubusercontent.com/72724939/187660895-142b36a3-552f-4835-a3bd-742ee1d86c27.png)

**Reply by lbyupup** [reply]
> 按照 @fz6m 提示的配置后报了下面这个错误，有大佬帮忙看看吗 ![image](https://user-images.githubusercontent.com/72724939/187660895-142b36a3-552f-4835-a3bd-742ee1d86c27.png)

已经解决。使用max的话 不需要加插件配置，只需要加model: {}

**Reply by WaterHong** [reply]
谢谢你，很有用，建议将这个错误写到升级文档里
